staging/src/k8s.io/apiserver/pkg/admission/plugin/initialization/initialization.go

```diff
@@ -296,9 +297,20 @@ func newInitializers(names []string) *metav1.Initializers {
 
 // findInitializers returns the list of initializer names that apply to a config. It returns an empty list
 // if no initializers apply.
-func findInitializers(initializers *v1alpha1.InitializerConfiguration, gvr schema.GroupVersionResource) []string {
+func findInitializers(initializers *v1alpha1.InitializerConfiguration, gvr schema.GroupVersionResource, namespace string) []string {
 	var names []string
 	for _, init := range initializers.Initializers {
+		if init.NamespaceSelector != nil {
+			selector, err := metav1.LabelSelectorAsSelector(init.NamespaceSelector)
+			if err != nil {
+				continue
+			}
+
+			if selector.Empty() || !selector.Matches(labels.Set(map[string]string{"namespace": namespace})) {
```

Comment:
Calling this a label selector when it is not actually selecting labels is misleading

Reply:
+1 I had the same reaction. It's unclear to me still whether this is a namespace name whitelist, or an actual namespace labelSelector.